packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -191,3 +186,86 @@ function isSecondsPropertyMisusedInCalendarTriggerInput(
       Object.keys(triggerWithoutChannelId).length > 1)
   );
 }
+
+function validateDateComponents(
+  object: any,
+  components: {
+    day?: boolean;
+    month?: boolean;
+    weekday?: boolean;
+    hour?: boolean;
+    minute?: boolean;
+  }
+) {
+  const expectedProperties = Object.keys(components).filter(key => components[key]);
+  if (
+    expectedProperties.some(
+      component => object[component] === undefined || object[component] === null
+    )
+  ) {
+    const headParams = [...expectedProperties];
+    const tailParam = headParams.pop();
+    const paramsText = headParams.length ? `${headParams.join(', ')} and ${tailParam}` : tailParam;
+    const plural = expectedProperties.length;
+    throw new TypeError(
+      `Parameter${plural ? 's' : ''} ${paramsText} need${
+        plural ? '' : 's'
+      } to have valid values. Found undefined`
+    );
+  }
+  if (components.month) {
+    const { month } = object;
+    if (month < 0 || month > 11) {
+      throw new RangeError(`The month parameter needs to be between 0 and 11. Found: ${month}`);
+    }
+  }
+  if (components.day) {
+    const { day, month } = object;
+    const daysInGivenMonth = daysInMonth(month);
+    if (day < 1 || day > daysInGivenMonth) {
+      throw new RangeError(
+        `The day parameter needs to be between 1 and the amount of days for the given month (${daysInGivenMonth} for month ${object.month}). Found: ${day}`
+      );
+    }
+  }
+  if (components.weekday) {
+    const { weekday } = object;
+    if (weekday < 1 || weekday > 7) {
+      throw new RangeError(`The weekday parameter needs to be between 1 and 7. Found: ${weekday}`);
+    }
+  }
+  if (components.hour) {
+    const { hour } = object;
+    if (hour < 0 || hour > 23) {
+      throw new RangeError(`The hour parameter needs to be between 0 and 23. Found: ${hour}`);
+    }
+  }
+  if (components.minute) {
+    const { minute } = object;
+    if (minute < 0 || minute > 59) {
+      throw new RangeError(`The minute parameter needs to be between 0 and 59. Found: ${minute}`);
+    }
+  }
+}
+
+/**
+ * Determines the number of days in the given month. If year is specified, it will include
+ * leap year logic, else it will always assume a leap year
+ */
+function daysInMonth(month: number, year?: number) {
+  switch (month) {
+    case 1:
+      return year === undefined || isLeapYear(year) ? 29 : 28;
+    case 3:
+    case 5:
+    case 8:
+    case 10:
+      return 30;
+    default:
+      return 31;
+  }
```

Comment:
```suggestion
  new Date(year ?? 2000, month, 0).getDate();
```

I would test this on android and iOS just to be sure- sometimes the Date functions a little differently between the two

Reply:
Nice trick using 0 to get the max date, although it would need to be more complicated than that, since that will return you the max day of the previous month… so it would need to be `new Date(year ?? 2000, (month + 1) % 12, 0).getDate();`… a nice one-liner but a bit cryptic semantically speaking in my opinion (and likely, although not importantly, less efficient because of Date creation). I don't mind making this change though, but please can you reconfirm?

Reply:
Yeah, I think the one line is better since it's a more standard way of getting days in a month in JS

is the `% 12` necessary here? `new Date()` should handle cases of month > 12

Reply:
yes, you are correct, Date also handles the month 12 case, turning it into December of year 1999 (which it doesn't matter, since December is not affected by leap years). I will use that.